pipeline.go

```diff
@@ -107,7 +82,20 @@ func stepsToTrigger(files []string, watch []WatchConfig) ([]Step, error) {
 					return nil, err
 				}
 				if match {
-					steps = append(steps, w.Step)
+					out, err := executeCommand("bash", []string{"-c", w.Generator})
+					if err != nil {
+						return nil, err
+					}
+
+					p, err := pipeline.Parse(strings.NewReader(out))
+					if err != nil {
+						return nil, err
+					}
+
+					for _, step := range p.Steps {
+						steps = append(steps, step)
+					}
+
```

Comment:
Here's the magic -- instead of expecting a hardcoded set of steps, we run the `generator` command, parse the output, and append those steps to the dynamic pipeline.